apps/src/code-studio/components/progress/StageLockDialog.jsx

```diff
@@ -144,6 +147,9 @@ const StageLockDialog = React.createClass({
     const responsiveHeight = {
       maxHeight: window.innerHeight * 0.8 - 100
     };
+    const hiddenUnlessSelectedSection = this.props.selectedSectionId === '' ?
+      styles.hidden : {};
+    const hasSelectedSection = this.props.selectedSectionId !== "";
```

Comment:
If you reversed these two statements you could define one in terms of the other.